app/javascript/controllers/validations_controller.js

```diff
@@ -0,0 +1,66 @@
+import Rails from '@rails/ujs'
+import Turbolinks from 'turbolinks'
+import debounce from 'lodash.debounce'
+import ApplicationController from './application_controller'
+
+export default class extends ApplicationController {
+  static targets = ['firstName', 'lastName', 'email', 'submit']
+
+  connect () {
+    super.connect()
+    this.perform = debounce(this._perform, 250)
+    this.firstNameTarget.focus()
```

Comment:
Seems like the correct way to address this is to put `autofocus: true` on the firstName element in the template, no?

I actually tried this and it seemed to work. Is there an edge case I'm not anticipating?

Reply:
I was trying to circumvent the autogenerated default autofocus behavior; however, I've rewritten that logic so yes... setting autofocus will work now.